dockerfiles/Dockerfile.training

```diff
@@ -0,0 +1,200 @@
+RUN apt-get -y update && apt-get -y --no-install-recommends install \
+        libibverbs-dev \
+        libnuma-dev &&\
+    cd /stage && curl -fSsL ${UCX_URL} | tar xzf - &&\
+    cd ${UCX_TARNAME} &&\
+    ./configure \
+	--prefix=/opt/ucx \
+        --with-cuda=/usr/local/cuda \
+        --with-verbs=/usr/lib/x86_64-linux-gnu \
+        --enable-mt &&\
+    make -j"$(nproc)" &&\
+    make install
+
+# build openmpi (use --prefix /opt/openmpi-xxx to move to runtime image)
+# note: require --enable-orterun-prefix-by-default for Azure machine learning compute
+# note: disable verbs as we use ucx middleware and don't want btl openib warnings
+ARG OPENMPI_VERSION
+ARG OPENMPI_TARNAME=openmpi-${OPENMPI_VERSION}
+ARG OPENMPI_URL=https://download.open-mpi.org/release/open-mpi/v%OMPI_BASE%/${OPENMPI_TARNAME}.tar.gz
+RUN export OMPI_BASE=${OPENMPI_VERSION%.*} &&\
+    cd /stage && curl -fSsL `echo ${OPENMPI_URL} | sed s/%OMPI_BASE%/$OMPI_BASE/` | tar xzf - &&\
+    cd ${OPENMPI_TARNAME} &&\
+    ./configure \
+        --prefix=/opt/openmpi-${OPENMPI_VERSION} \
+        --with-ucx=/opt/ucx \
+        --without-verbs \
+        --with-cuda=/usr/local/cuda \
+        --enable-mpirun-prefix-by-default \
+        --enable-orterun-prefix-by-default \
+        --enable-mca-no-build=btl-uct &&\
+    make -j"$(nproc)" install &&\
+    ldconfig
+ENV PATH=/opt/openmpi-${OPENMPI_VERSION}/bin:$PATH
+ENV LD_LIBRARY_PATH=/opt/openmpi-${OPENMPI_VERSION}/lib:$LD_LIBRARY_PATH
+
+# install mpi4py (be sure to link existing /opt/openmpi-xxx)
+RUN CC=mpicc MPICC=mpicc pip install mpi4py --no-binary mpi4py
+
+# install pytorch
+ARG PYTORCH_VERSION
+RUN pip install torch==${PYTORCH_VERSION}
+
+# in case you need to build pytorch:
+# note: if you want specific branch or to link system cuda libraries or MPI
+# note: recommend using many high-frequency cores (e.g. 32+ skylake cores)
+# ENV CUDA_HOME="/usr/local/cuda" \
+#     CUDNN_LIBRARY="/usr/lib/x86_64-linux-gnu" \
+#     NCCL_INCLUDE_DIR="/usr/include" \
+#     NCCL_LIB_DIR="/usr/lib/x86_64-linux-gnu" \
+#     USE_SYSTEM_NCCL=1 
+# RUN conda install -y \
+#         mkl \
+#         mkl-include \
+#         ninja \
+#         pyyaml \
+#         cffi &&\
+#     cd /stage && git clone https://github.com/pytorch/pytorch.git &&\
+#     cd pytorch &&\
+#     git checkout v1.6.0 &&\
+#     git submodule update --init --recursive &&\
+#     python setup.py bdist_wheel -d build/wheel &&\
+#     pip install build/wheel/*.whl
+
+# build onnxruntime wheel with cuda and mpi support
+RUN cd /stage && git clone https://github.com/microsoft/onnxruntime.git &&\
+    cd onnxruntime &&\
+    git checkout master &&\
+    cp ThirdPartyNotices.txt /stage/ThirdPartyNotices.txt &&\
+    cp dockerfiles/LICENSE-IMAGE.txt /stage/LICENSE-IMAGE.txt &&\
+    python tools/ci_build/build.py \
+        --cmake_extra_defines \
+            ONNXRUNTIME_VERSION=`cat ./VERSION_NUMBER` \
+        --config Release \
```

Comment:
Recommend moving 'Release' to a `ARG BUILD_VERSION=Release` clause in dockerfile for flexbility